pystac/validation/__init__.py

```diff
@@ -0,0 +1,109 @@
+# flake8: noqa
+import pystac
+from pystac.serialization.identify import identify_stac_object
+
+from pystac.validation.schema_uri_map import (SchemaUriMap)
+
+
+class STACValidationError(Exception):
+    """Represents a validation error. Thrown by validation calls if the STAC JSON
+    is invalid.
+
+    Args:
+        source (object): Source of the exception. Type will be determined by the
+            validation implementation. For the default JsonSchemaValidator this will a
+            the ``jsonschema.ValidationError``.
+    """
+    def __init__(self, message, source=None):
+        super().__init__(message)
+        self.source = source
+
+
+# Import after above class definition
+from pystac.validation.stac_validator import (STACValidator, JsonSchemaSTACValidator)
+
+
+def validate_json(stac_json_dict):
+    """Validate STAC json.
+
+    Raises a STACValidationError if invalid; otherwise runs and returns nothing.
+
+    Args:
+        stac_json_dict (dict): The JSON of the STAC object.
+
+    Raises:
+        STACValidationError
+    """
+    info = identify_stac_object(stac_json_dict)
+    return validate(stac_json_dict,
+                    stac_object_type=info.object_type,
+                    stac_version=info.version_range.latest_valid_version(),
+                    extensions=info.common_extensions)
+
+
+def validate(stac_dict, stac_object_type, stac_version=None, extensions=None):
+    """Validate a stac object based on the object type, stac version, and a list of
+    implementing extensions.
+
+    This method delegates to the call to :meth:`pystac.validation.STACValidator.validate`
+    for the STACValidator registered via :meth:`~pystac.validation.set_validator` or
+    :class:`~pystac.validation.JsonSchemaSTACValidator` by default.
+
+    Args:
+        stac_dict (dict): Dictionary that is the STAC json of the object.
+        stac_object_type (str): The stac object type of the object encoded in stac_dict.
+            One of :class:`~pystac.STACObjectType`.
+        stac_version (str): The version of STAC to validate the object against.
+        extensions (List[str]): Extension IDs for this stac object.
```

Comment:
Consider combining theses two functions: allowing `stac_object_type` to be none and infer from dict. Change name to `validate_dict()`